ci: Switch to dynamic library linkage in native Windows job

hebasto opened the pull request:
This PR significantly reduces the vcpkg binary cache size, improving CI caching performance:

| Branch | Cache Size |
|---|--:|
| master | 2.6 GB |
| this PR |  430 MB |

Also see https://github.com/bitcoin/bitcoin/pull/31176#issuecomment-2766164288.

hebasto commented:
> Was there a reason to use static linking in this CI here before?

The static linking was inherited from the Cirrus task in https://github.com/bitcoin/bitcoin/pull/28173, which, in turn, used our custom Qt build instead of the vcpkg manager.

My guess is that, back when we were using our custom MSVC project file, static linking worked seamlessly without needing to manage DLL locations.

Maybe @sipsorcery could provide more details.

sipsorcery commented:
The msvc build was created with static linking because that's what the non-msvc build does. Other than that no real reason.

hebasto committed:
ci: Switch to dynamic library linkage in native Windows job

This change significantly reduces the vcpkg binary cache size, improving
CI caching performance.

hebasto commented:
Rebased and undrafted.
